encoder.lua

```diff
@@ -152,36 +186,38 @@ mpv.make_audio_args = function(source_path, output_path, start_timestamp, end_ti
     }
 end
 
+
 ------------------------------------------------------------
--- main interface
 
-local create_snapshot = function(timestamp, filename)
-    if not h.is_empty(self.config.image_field) then
-        local source_path = mp.get_property("path")
-        local output_path = utils.join_path(self.platform.tmp_dir(), filename)
-        local on_finish = function()
-            self.store_fn(filename, output_path)
-            os.remove(output_path)
-        end
-        if not self.config.screenshot then
-            local args = self.encoder.make_snapshot_args(source_path, output_path, timestamp)
-            h.subprocess(args, on_finish)
-        else
-            local args = {'screenshot-to-file', output_path, 'video',}
-            mp.command_native_async(args, on_finish)
-        end
+-- Creates the animated snapshot and then calls on_finish_fn
+local create_animated_snapshot = function(start_timestamp, end_timestamp, source_path, output_path, on_finish_fn)
+    local args = ffmpeg.make_animated_snapshot_args(source_path, output_path, start_timestamp, end_timestamp)  -- ffmpeg is needed in order to generate animations
+    h.subprocess(args , on_finish_fn)
+end
+
+
+local create_snapshot = function(timestamp, source_path, output_path, on_finish_fn)
+    if not self.config.screenshot then
+        local args = self.encoder.make_snapshot_args(source_path, output_path, timestamp)
+        h.subprocess(args, on_finish_fn)
     else
-        print("Snapshot will not be created.")
+        local args = {'screenshot-to-file', output_path, 'video',}
+        mp.command_native_async(args, on_finish_fn)
     end
+
 end
 
+
 local background_play = function(file_path, on_finish)
     return h.subprocess(
-            { 'mpv', '--audio-display=no', '--force-window=no', '--keep-open=no', '--really-quiet', file_path },
-            on_finish
+        { 'mpv', '--audio-display=no', '--force-window=no', '--keep-open=no', '--really-quiet', file_path },
+        on_finish
```

Comment:
This change in indentation is redundant.

Reply:
Fixed